devicetypes/fibargroup/fibaro-motion-sensor-zw5.src/fibaro-motion-sensor-zw5.groovy

```diff
@@ -1,315 +1,478 @@
+*/
+def parse(String description) {
+    def result = []
+    logging("${device.displayName} - Parsing: ${description}")
+    if (description.startsWith("Err 106")) {
+        result = createEvent(
+                descriptionText: "Failed to complete the network security key exchange. If you are unable to receive data from it, you must remove it from your network and add it again.",
+                eventType: "ALERT",
+                name: "secureInclusion",
+                value: "failed",
+                displayed: true,
+        )
+    } else if (description == "updated") {
+        return null
+    } else {
+        def cmd = zwave.parse(description, cmdVersions())
+        if (cmd) {
+            logging("${device.displayName} - Parsed: ${cmd}")
+            zwaveEvent(cmd)
+        }
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand(cmdVersions())
+    if (encapsulatedCommand) {
+        logging("${device.displayName} - Parsed SecurityMessageEncapsulation into: ${encapsulatedCommand}")
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        log.warn "Unable to extract secure cmd from $cmd"
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+    def version = cmdVersions()[cmd.commandClass as Integer]
+    def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+    def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+    if (encapsulatedCommand) {
+        logging("${device.displayName} - Parsed Crc16Encap into: ${encapsulatedCommand}")
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        log.warn "Could not extract crc16 command from $cmd"
+    }
+}
+
+private logging(text, type = "debug") {
+    if (settings.logging == "true") {
+        log."$type" text
+    }
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+    logging("${device.displayName} - encapsulating command using Secure Encapsulation, command: $cmd","info")
+    zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+    logging("${device.displayName} - encapsulating command using CRC16 Encapsulation, command: $cmd","info")
+    zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format() // doesn't work righ now because SmartThings...
+    //"5601${cmd.format()}0000"
+}
+
+private encap(physicalgraph.zwave.Command cmd) {
+    if (zwaveInfo.zw.contains("s")) {
+        secEncap(cmd)
+    } else if (zwaveInfo.cc.contains("56")){
+        crcEncap(cmd)
+    } else {
+        logging("${device.displayName} - no encapsulation supported for command: $cmd","info")
+        cmd.format()
+    }
+}
+
+private encapSequence(cmds, delay=250) {
+    delayBetween(cmds.collect{ encap(it) }, delay)
+}
+
+private List intToParam(Long value, Integer size = 1) {

```

Comment:
same comment as before. You should just be able to pass the value as `scaledConfigurationValue` instead of calling this here